src/runtime-rs/crates/hypervisor/src/device/driver/virtio_blk.rs

```diff
@@ -78,11 +82,20 @@ impl Device for BlockDevice {
         {
             return Ok(());
         }
-        if let Err(e) = h.add_device(DeviceType::Block(self.clone())).await {
-            self.decrease_attach_count().await?;
-            return Err(e);
+
+        match h.add_device(DeviceType::Block(self.clone())).await {
+            Ok(dev) => {
+                // Update device info with the one received from device attach
+                if let DeviceType::Block(blk) = dev {
+                    self.config = blk.config;
+                }
+                Ok(())
+            }
+            Err(e) => {
+                self.decrease_attach_count().await?;
```

Comment:
If this fails, we'll return the count error, not the original one (line below).

Reply:
This is the same as previous behaviour. If we want to change this, lets discuss. It could go as part of a separate change.